lib/utils/display.js

```diff
@@ -186,29 +219,27 @@ class Display {
   }
 
   #log (...args) {
-    return this.#eresolveWarn(...args) || this.#npmlog(...args)
-  }
-
-  // Explicitly call these on npmlog and not log shim
-  // This is the final place we should call npmlog before removing it.
-  #npmlog (level, ...args) {
-    npmlog[level](...args.map(Display.clean))
+    if (args.length === 4) {
```

Comment:
This seems very specific and has the code smell of a "footgun".

Reply:
I think I was trying to decide how to write this if condition and never finished. This length check should be removed and only the inner check needs to happen.

But hopefully we can finally unwind this hack after this refactor.